### PR TITLE
#686: Add `Contracts#remove(Contract)`

### PR DESCRIPTION
**Pull request by andreoss**
Per #686 
- Add method `remove`

### CODE REVIEW

**Comment by charlesmike** [comment]
@andreoss thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-11-17T17:00:31.585823``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-core-impl/src/main/java/com/selfxdsd/core/contracts/ContributorContracts.java [hunk 1]
```diff
@@ -175,4 +175,9 @@ public Contract markForRemoval(
     public Iterator<Contract> iterator() {
         return this.contracts.get().iterator();
     }
+
+    @Override
+    public void remove(final Contract contract) {
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/contracts/ContributorContracts.java:180]: @andreoss Similar to ProjectContracts.remove(...), we should first check if the given Contract belongs to the Contributor.

The username and the provider should be equal.

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@andreoss just one comment

File: self-core-impl/src/main/java/com/selfxdsd/core/contracts/ContributorContracts.java (hunk 1 above)
Comment (andreoss) [reply self-core-impl/src/main/java/com/selfxdsd/core/contracts/ContributorContracts.java:170]: @amihaiemil Done.

**Comment by charlesmike** [comment]
@amihaiemil Don't forget to close this ticket before the deadline (2020-11-17T17:00:31). You are past the first half of the allowed period.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**